src/app/credExplorer/pagerankTable/aggregate.test.js

```diff
@@ -1,7 +1,344 @@
+    };
+    const connections = {
+      loop5: {adjacency: adjacencies.loop5, weight: 0.2},
+      rfz4: {adjacency: adjacencies.rfz4, weight: 0.2},
+      rfk1: {adjacency: adjacencies.rfk1, weight: 0.2},
+      rbk2: {adjacency: adjacencies.rbk2, weight: 0.2},
+      kfr3: {adjacency: adjacencies.kfr3, weight: 0.2},
+    };
+    const scoredConnections = {
+      loop5: {
+        connection: connections.loop5,
+        source: nodes.root,
+        sourceScore: 0.2,
+        connectionScore: 5,
+      },
+      rfz4: {
+        connection: connections.rfz4,
+        source: nodes.zap,
+        sourceScore: 0.2,
+        connectionScore: 4,
+      },
+      kfr3: {
+        connection: connections.kfr3,
+        source: nodes.kif,
+        sourceScore: 0.2,
+        connectionScore: 3,
+      },
+      rbk2: {
+        connection: connections.rbk2,
+        source: nodes.kif,
+        sourceScore: 0.2,
+        connectionScore: 2,
+      },
+      rfk1: {
+        connection: connections.rfk1,
+        source: nodes.kif,
+        sourceScore: 0.2,
+        connectionScore: 1,
+      },
+    };
+    const scoredConnectionsArray = [
+      scoredConnections.loop5,
+      scoredConnections.rfz4,
+      scoredConnections.rfk1,
+      scoredConnections.rbk2,
+      scoredConnections.kfr3,
+    ];
+    return {
+      nodes,
+      edges,
+      nodeTypes,
+      nodeTypesArray,
+      edgeTypes,
+      edgeTypesArray,
+      adjacencies,
+      connections,
+      scoredConnections,
+      scoredConnectionsArray,
+    };
+  }
+  describe("aggregateByNodeType", () => {
+    function exampleNodeAggregation() {
+      const {nodeTypesArray, scoredConnectionsArray} = example();
+      const aggregations = aggregateByNodeType(
+        scoredConnectionsArray,
+        nodeTypesArray
+      );
+      return aggregations;
+    }
+    it("puts every connection in an aggregation", () => {
+      const {scoredConnectionsArray} = example();
+      let connectionsFound = 0;
+      for (const aggregation of exampleNodeAggregation()) {
+        connectionsFound += aggregation.connections.length;
+      }
+      expect(connectionsFound).toEqual(scoredConnectionsArray.length);
+    });
+    it("groups connections by node type", () => {
+      for (const aggregation of exampleNodeAggregation()) {
+        for (const connection of aggregation.connections) {
+          expect(
+            NodeAddress.hasPrefix(
+              connection.source,
+              aggregation.nodeType.prefix
+            )
+          ).toBe(true);
+        }
+      }
+    });
+    it("when multiple node types match, it uses the most specific one", () => {
+      const {nodeTypes, scoredConnections} = example();
+      const aggregations = aggregateByNodeType(
+        [scoredConnections.rfz4],
+        [nodeTypes.empty, nodeTypes.zap]
+      );
+      expect(aggregations).toHaveLength(1);
+      // It used the zap nodeType rather than the empty nodeType b/c it was more specific
```

Comment:
nit: please wrap (preferably at 72, but 80 is okay)

Reply:
Done